src/features/tags/components/TagsList.tsx

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { Tag } from "../types";
+import TagItem from "./TagItem";
+
+type TagsListProps = {
+  tags: Tag[] | undefined;
```

Comment:
So my misnamed functions messed you up here. The tags prop here is actually an array of tag ids, not the full tag object. This affects your code in a few other places and it's because I misnamed a few selector functions. So this type is `number[] | undefined`. This prop should probably be renamed to `tagIds` or just `ids`